framework/Models/Code.py

```diff
@@ -513,10 +513,17 @@ def evaluateSample(self, myInput, samplerType, kwargs):
 
     ## If the run was successful
     if returnCode == 0:
-
       returnDict = {}
+      ## Special case for RAVEN interface --ALFOA 09/17/17
+      ravenCase = False
+      if type(outputFile).__name__ == 'dict':
+        ravenCase = True
+      if ravenCase and self.code.__class__.__name__ != 'RAVEN':
+        self.raiseAnError(RuntimeError, 'The return argument from "finalizeCodeOutput" must be a str containing the new output file root!')
```

Comment:
This basically forbids any code besides RAVEN from returning a dictionary.  I am trying to decided if this is good or bad.

Reply:
yes I did on purpose to avoid any other code interface to return a different thing than a outputfilename root... 

I thinks we should allow to do that but the system should be better designed...